plasmapy/analytic/parameters.py

```diff
@@ -263,13 +294,25 @@ def electron_thermal_speed(T_e):
 
     try:
         T_e = T_e.to(u.K, equivalencies=u.temperature_energy())
+    except Exception:
+        raise u.UnitConversionError("The electron temperature in "
+                                    "electron_thermal_speed "
+                                    "cannot be converted to Kelvin.")
+
+    if np.any(not 0 <= T_e.value <= np.inf):
+        raise ValueError("Invalid temperature in electron_thermal_speed.")
+
+    try:    
         V_Te = (np.sqrt(2*k_B*T_e/m_e)).to(u.m/u.s)
     except Exception:
         raise ValueError("Cannot find electron thermal speed")
 
     if V_Te > c:
         raise UserWarning("Electron thermal speed is greater than speed "
-                          "of light")
+                          "of light.")
+    elif V_Te > 0.1*c:
+        raise UserWarning("Electron thermal speed is greater than 10% of "
+                          "speed of light.")
```

Comment:
This is one of the things I still am uncertain about.  I personally think a warning like this would be worthwhile since most of these expressions lose validity as relativistic effects become important, but I am not certain what fraction of the speed of light would be most appropriate.  For 0.1c, the Lorentz factor is ~1.005, for 0.2c, it's ~1.02, and for 0.3c it's ~1.05.  I doubt that we would have accuracy better than ~1-10% anyway in most plasma situations anyway so maybe 0.2c or 0.3c would be appropriate as a cutoff for the warning, but 0.1c is what I saw most often elsewhere so I defaulted to that.  I'm flexible on this in case anyone else has other opinions!

Come to think of it, somewhere we should have a function for the Lorentz factor.  It may be useful to include that information in the warning message.